daemon/state.go

```diff
@@ -371,7 +371,7 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 	}
 
 	for _, ep := range state.toClean {
-		go d.deleteEndpointQuiet(ep, true)
+		d.deleteEndpointQuiet(ep, true)
```

Comment:
Deleting endpoints could potentially take a while since there are kvstore interactions involved when it is being deleted. What about the following?

```
var deleteEpsWG sync.WaitGroup
...
for _, ep := range state.toClean {
  deleteEpsWG.Add(1)
  go func(ep *endpoint.Endpoint) {
    d.deleteEndpointQuiet(ep, true)
    deleteEpsWG.Done()
  }(ep)
}
... 
deleteEpsWG.Wait()
```

I'd also add a comment here indicating why we are doing this synchronously in the code itself (it can be the same as the commit message).